sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/implementation/NonAzureOpenAIKeyCredentialPolicy.java

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.openai.implementation;
+
+import com.azure.ai.openai.models.NonAzureOpenAIKeyCredential;
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpPipelineNextSyncPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+
+import java.util.Objects;
+
+/**
+ * Pipeline policy that uses an {@link NonAzureOpenAIKeyCredential} to set the authorization key for a request to
+ * an HTTP request.
+ * <p>
+ * Requests sent with this pipeline policy are required to use {@code HTTPS}. If the request isn't using {@code HTTPS}
+ * an exception will be thrown to prevent leaking the key.
+ */
+public class NonAzureOpenAIKeyCredentialPolicy implements HttpPipelinePolicy {
```

Comment:
Since the AzureKeyCredentialPolicy is public, can we reuse it here instead of duplicating the whole class?

Reply:
I was thinking about the same idea. but with the "Azure" prefix for NonAzure sounds wrong.

Reply:
The policy will require the NonAzureOpenAIKeyCredenial type. And this type is for  NonAzure requests. We should not mix the concept of Azure and NonAzure credentials.